hedy.py

```diff
@@ -1889,13 +1890,19 @@ def make_print_ask_arg(self, arg, meta, var_to_escape=''):
         # therefore we should not process it anymore and thread it as a variable:
         # we set the line number to 100 so there is never an issue with variable access before
         # assignment (regular code will not work since random.choice(dieren) is never defined as var as such)
-        if "random.choice" in arg or "[" in arg:
-            return self.process_variable_for_fstring(arg, meta.line, var_to_escape)
-
-        # this regex splits words from non-letter characters, such that name! becomes [name, !]
-        p = r"[·\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Lo}\p{Nl}\p{Mn}\p{Mc}\p{Nd}\p{Pc}]+|[^·\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Lo}\p{Nl}]+"
-        res = regex.findall(p, arg)
-        return ''.join([self.process_variable_for_fstring(x, meta.line, var_to_escape) for x in res])
+        if not self.microbit:
+            if "random.choice" in arg or "[" in arg:
+                return self.process_variable_for_fstring(arg, meta.line, var_to_escape)
+            # this regex splits words from non-letter characters, such that name! becomes [name, !]
+            p = r"[·\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Lo}\p{Nl}\p{Mn}\p{Mc}\p{Nd}\p{Pc}]+|[^·\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Lo}\p{Nl}]+"
+            res = regex.findall(p, arg)
+            return ''.join([self.process_variable_for_fstring(x, meta.line, var_to_escape) for x in res])
+        else:
+            if self.is_variable(arg, meta.line):
+                return arg
+            else:
+                # If the argument is not a variable, return it as a string literal with quotes
+                return f"'{arg}'"
```

Comment:
If the argument contains a single quote, this code would not transpile correctly. Luckily, we already have a function that escapes single quotes. Consider changing this line to:
`return f"'{process_characters_needing_escape(arg)}'"`